### PR TITLE
test(query): added query unit tests

### PR DESCRIPTION
**Pull request by YonatanKra**
note the TODO is a test we should decide if we want it or not (#312 )

### CODE REVIEW

File: src/askvm/resources/core/__test__/query.test.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { runUntyped } from '../../..';
+import { parse } from '../../../../askcode';
```
Comment (mhagmajer) [inline src/askvm/resources/core/__test__/query.test.ts:2]: I wonder if for resource unit test we should be calling the resource directly or keep going through the AskCode parser like you're doing here 🤔 . Maybe it would be easier to catch any potential problems and check for different scenarios if we were calling `query` resource directly. What do you guys think @YonatanKra @czerwinskilukasz1?

Comment (YonatanKra) [reply src/askvm/resources/core/__test__/query.test.ts:2]: I didn't even notice it. Just my IDE auto import doing its thing.
Definitely better to do direct here. The post build tests willmake sure it is available in the barrel file.

Comment (YonatanKra) [reply src/askvm/resources/core/__test__/query.test.ts:2]: Looking closer at your response I now see you mean to test the function directly. 
Should make for easier testing.

I'll try that.

I just took existing tests and followed conventions in this area.

Both ways work for me as long as I can test. 

Main point is to decide on good conventions and stick to them so read8ng one part of code would look similar to others (in regards to style etc.).

Comment (mhagmajer) [reply src/askvm/resources/core/__test__/query.test.ts:2]: @YonatanKra yes, you are right that we're missing consistency there. I think personally that low level unit test for a function should just test this function and not rely to much on other code - this way they are more resilient and allow for testing a greater variety of edge cases. We also have this other kind of tests that are more like "unit integration tests" where we checking results for askScript programs.

Either way both approaches here are acceptable.

File: src/askvm/resources/core/__test__/query.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  //   const expectedObjectValue = {
+  //     objectReturned: values.objectValue,
+  //   };
+  //
+  //   const queryForAnObjectValue = `query(node('objectReturned',f(object('a',call(get('get'),'objectValue')))))`;
+  //
+  //   await expect(ask(queryForAnObjectValue)).resolves.toStrictEqual(
+  //     expectedObjectValue
+  //   );
+  // });
```
Comment (mhagmajer) [inline src/askvm/resources/core/__test__/query.test.ts:92]: Instead of having commented code here can we link to a GitHub issue in a comment and maybe have that code there?

**Review by mhagmajer — COMMENTED** [review]
@YonatanKra almost there! Please add a test case of calling `query` resolver directly if you think that would be valuable

File: src/askvm/resources/core/__test__/query.test.ts (hunk 2 above)
Comment (czerwinskilukasz1) [reply src/askvm/resources/core/__test__/query.test.ts:92]: @YonatanKra I am fine with leaving the code there, but please please add a link to the GitHub issue.

Comment (YonatanKra) [reply src/askvm/resources/core/__test__/query.test.ts]: I don't like comments in code - especially when we have git :)
So I've copied the code to the issue itself.

**Review by mhagmajer — APPROVED** [review]
👏
